crates/sui-indexer-alt-graphql/src/api/types/execution_error.rs

```diff
@@ -125,6 +180,103 @@ impl ExecutionError {
+                    write_msg(&mut msg, format_args!("{}", self.native))?;
+                    return Ok(msg);
+                };
+
+                // Format Move abort with module and function info
+                write_msg(&mut msg, format_args!("from '{}", loc.module.to_canonical_display(true)))?;
+                if let Some(fname) = &loc.function_name {
+                    write_msg(&mut msg, format_args!("::{}'", fname))?;
+                } else {
+                    write_msg(&mut msg, format_args!("'"))?;
+                }
+
+                // Try to get clever error information
+                let Some(CleverError {
+                    source_line_number,
+                    error_info,
+                    error_code,
+                    ..
+                }) = self.clever_error().await.as_ref()
+                else {
+                    // No clever error, show basic abort info
+                    write_msg(
+                        &mut msg,
+                        format_args!(" (instruction {}), abort code: {code}", loc.instruction),
+                    )?;
+                    return Ok(msg);
+                };
+
+                // Format with clever error details
+                let error_code_str = match error_code {
+                    Some(code) => format!("(code = {code})"),
+                    _ => String::new(),
+                };
+
+                match error_info {
+                    ErrorConstants::Rendered { identifier, constant } => {
+                        write_msg(
+                            &mut msg,
+                            format_args!(" (line {source_line_number}), abort{error_code_str} '{identifier}': {constant}"),
+                        )?;
+                    }
+                    ErrorConstants::Raw { identifier, bytes } => {
+                        let const_str = Base64::encode(bytes);
+                        write_msg(
+                            &mut msg,
+                            format_args!(" (line {source_line_number}), abort{error_code_str} '{identifier}': {const_str}"),
+                        )?;
+                    }
+                    ErrorConstants::None => {
+                        write_msg(
+                            &mut msg,
+                            format_args!(
+                                " (line {source_line_number}){}",
+                                match error_code {
+                                    Some(code) => format!(" abort(code = {code})"),
+                                    _ => String::new(),
+                                }
+                            ),
+                        )?;
+                    }
+                }
+
+                Ok(msg)
+            }
+        }
+    }
+}
+
+/// Helper function to write to a string with proper error context.
+fn write_msg(msg: &mut String, args: std::fmt::Arguments) -> Result<(), RpcError> {
```

Comment:
nit: you can avoid this helper and just use `write!` directly in `format_message` if you change its return type to return whatever error `write!` produces (`io::Error` I think), then chain a single call to `.context()` on the call to `format_message` on L133.